envs/monkey_zoo/blackbox/tests/performance/endpoint_performance_test.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from datetime import timedelta
+
+from envs.monkey_zoo.blackbox.tests.basic_test import BasicTest
+from envs.monkey_zoo.blackbox.island_client.monkey_island_client import MonkeyIslandClient
+from envs.monkey_zoo.blackbox.tests.performance.performance_test_config import PerformanceTestConfig
+from envs.monkey_zoo.blackbox.analyzers.performance_analyzer import PerformanceAnalyzer
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class EndpointPerformanceTest(BasicTest):
+
+    def __init__(self, name, test_config: PerformanceTestConfig, island_client: MonkeyIslandClient):
+        self.name = name
+        self.test_config = test_config
+        self.island_client = island_client
+
+    def run(self) -> bool:
+        if not self.island_client.is_all_monkeys_dead():
+            raise RuntimeError("Can't test report times since not all Monkeys have died.")
+
+        # Collect timings for all pages
+        self.island_client.clear_caches()
+        endpoint_timings = {}
+        for endpoint in self.test_config.endpoints_to_test:
+            endpoint_timings[endpoint] = self.get_elapsed_for_get_request(endpoint)
+
+        analyzer = PerformanceAnalyzer(self.test_config, endpoint_timings)
```

Comment:
Unclear on why this isn't passed in the "Analyzers" parameter of the c'tor

Reply:
Because we don't know `endpoint_timings` before running the test. If we knew that all analyzers will take only these two parameters we could pass it on the constructor and then iterate on analyzers, but since there's only one analyzer used it doesn't make sense to refactor for now.